src/main/java/ch/naviqore/service/impl/StopTimeImpl.java

```diff
@@ -10,13 +10,14 @@
 @AllArgsConstructor(access = AccessLevel.PACKAGE)
 @RequiredArgsConstructor(access = AccessLevel.PACKAGE)
 @Getter
-@ToString
+@ToString(exclude = "trip")
 public class StopTimeImpl implements StopTime {
 
     private final Stop stop;
     private final LocalDateTime arrivalTime;
     private final LocalDateTime departureTime;
     @Setter(AccessLevel.PACKAGE)
-    private Trip trip;
+    // cyclical dependency, therefore avoid serialization and use in toString representation
+    private transient Trip trip;
```

Comment:
More out of interest, the transient is for serialization and the tostring exclude is only for tostring?

Reply:
Yes, I thought that transient would be also ignored by Lombok, but was not the case.
We could open a feature request there, since I think it makes no sense to have something in the toString but not in serialize.